designer/server/src/main/scala/pl/touk/nussknacker/ui/process/newdeployment/DeploymentService.scala

```diff
@@ -0,0 +1,145 @@
+
+import cats.data.EitherT
+import db.util.DBIOActionInstances._
+import pl.touk.nussknacker.engine.api.deployment.DataFreshnessPolicy
+import pl.touk.nussknacker.engine.api.deployment.StateStatus.StatusName
+import pl.touk.nussknacker.engine.api.process.{ProcessIdWithName, ProcessName}
+import pl.touk.nussknacker.security.Permission
+import pl.touk.nussknacker.ui.db.entity.ProcessEntityData
+import pl.touk.nussknacker.ui.process.deployment.{
+  CommonCommandData,
+  DeploymentService => LegacyDeploymentService,
+  RunDeploymentCommand => LegacyRunDeploymentCommand
+}
+import pl.touk.nussknacker.ui.process.newdeployment.DeploymentEntityFactory.DeploymentEntityData
+import pl.touk.nussknacker.ui.process.newdeployment.DeploymentService._
+import pl.touk.nussknacker.ui.process.repository.{CommentValidationError, DBIOActionRunner, ScenarioMetadataRepository}
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import java.sql.Timestamp
+import java.time.Clock
+import scala.concurrent.{ExecutionContext, Future}
+
+// TODO: This class is a new version of deployment.DeploymentService. The problem with the old one is that
+//       it joins multiple responsibilities like activity log (currently called "actions") and deployments management.
+//       Also, because of the fact that periodic mechanism is build as a plug-in (DeploymentManager), some deployment related
+//       operations (run now operation) is modeled as a CustomAction. Eventually, we should:
+//       - Split activity log and deployments management
+//       - Move periodic mechanism into to the designer's core
+//       - Remove CustomAction
+//       After we do this, we can remove legacy classes and fully switch to the new once.
+class DeploymentService(
+    scenarioMetadataRepository: ScenarioMetadataRepository,
+    deploymentRepository: DeploymentRepository,
+    // TODO: we shouldn't call legacy service, instead we should new service from the legacy one
+    legacyDeploymentService: LegacyDeploymentService,
+    dbioRunner: DBIOActionRunner,
+    clock: Clock
+)(implicit ec: ExecutionContext) {
+
+  def processCommand(
+      command: DeploymentCommand
+  ): Future[Either[RunDeploymentError, Unit]] = {
+    command match {
+      case command: RunDeploymentCommand =>
+        runDeployment(command)
+    }
+  }
+
+  private def runDeployment(
+      command: RunDeploymentCommand
+  ): Future[Either[RunDeploymentError, Unit]] =
+    (for {
+      scenarioMetadata <- getScenarioMetadata(command)
+      _ <- EitherT.cond[Future](
+        command.user.can(scenarioMetadata.processCategory, Permission.Deploy),
+        (),
+        NoPermissionError
+      )
+      _         <- saveDeployment(command, scenarioMetadata)
+      runResult <- invokeLegacyRunDeploymentLogic(command, scenarioMetadata)
+    } yield runResult).value
+
+  private def getScenarioMetadata(command: RunDeploymentCommand) =
+    EitherT[Future, RunDeploymentError, ProcessEntityData](
```

Comment:
I mentioned great factory methods of the EitherT. Take a look:
```scala
    EitherT.fromOptionF(
      fopt = scenarioMetadataRepository.getScenarioMetadata(command.scenarioName),
      ifNone = ScenarioNotFoundError(command.scenarioName)
    )
```

Reply:
Yeah I already used it in one place, now it is used here as well